experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1571,19 +1590,28 @@ def _validate_feature_value_against_schema(
                         continue
 
                     if schema_errors := self._validate_schema(
-                        substituted_value, json_schema, schema.version
+                        substituted_value, json_schema
                     ):
                         err_msg = (
                             f"Schema validation errors occured during locale "
                             f"substitution for locale {locale_code}"
                         )
 
-                        if schema.version is not None:
-                            err_msg += f" at version {schema.version}"
-
-                        errors.append(err_msg)
-                        errors.extend(schema_errors)
-
+                        if version is not None:
+                            errs_with_version = self.add_to_versioned_errs(
+                                # covered by test_validate_feature_versioned_localized
+                                [*schema_errors, err_msg], # pragma: no cover
+                                version,
+                                errs_with_version,
+                            )
+                        else:
+                            errors.append(err_msg)
+                            errors.extend(schema_errors)
+
+        if len(errs_with_version.keys()) > 0:
+            formatted_errs = self._format_grouped_errors(errs_with_version)
+            for e in formatted_errs:
+                errors.append(e)
```

Comment:
You can use [`extend()`](https://docs.python.org/3/library/stdtypes.html#mutable-sequence-types) here:

```suggestion
            errors.extend(self._format_grouped_errors(errs_with_version)
```

However, I think we should return `dict[str, list[NimbusFeatureversion]]` from both the FML and Schema validators, and then we can call the format function in a single place, e.g.


```python
def _validate_feature_configs(self, data):
    # ...
    for feature_value_data in data.get("reference_branch", {}).get("feature_values", []):
        if application == NimbusExperiment.Application.DESKTOP:
            if schema_errors := self._validate_feature_value_against_schema(...):
                reference_branch_errors.append({"value": self._format_grouped_errors(schema_errors) })
        elif NimbusExperiment.Application.is_mobile(application): # NB <- This condition is wrong now! we aren't validating cirrus with fml!
            if fml_errors := self._validate_with_fml(...):
                reference_branch_errors.append({"value": self._format_grouped_errors(fml_errors) })     
```

Reply:
That's what I was thinking too (re: moving these into the top level "if desktop, format errors/ if not, format errors")